ApplyToBecomeInternal/ApplyToBecomeInternal.Tests/Pages/ApplicationForm/ApplicationFormIntegrationTests.cs

```diff
@@ -257,5 +257,94 @@ public async void Should_Show_500Error_When_Application_Is_Not_Valid()
+		{
+			AddProjectWithFullApplicationForm();
+
+			await OpenUrlAsync($"/school-application-form/{_project.Id}");
+
+			Document.QuerySelectorAll("#Overview_link")
+				.Where(contents => contents.InnerHtml == "Overview")
+				.Should()
+				.NotBeEmpty();
+			Document.QuerySelectorAll("#Overview_link")
+				.First().Attributes.Where(a => a.Value == "#Overview")
+				.Should()
+				.NotBeEmpty();
+
+			Document.QuerySelectorAll("#About_the_conversion_link")
+				.Where(contents => contents.InnerHtml == "About the conversion")
+				.Should()
+				.NotBeEmpty();
+			Document.QuerySelectorAll("#About_the_conversion_link")
+				.First().Attributes.Where(a => a.Value == "#About_the_conversion")
+				.Should()
+				.NotBeEmpty();
+
+			Document.QuerySelectorAll("#Further_information_link")
+				.Where(contents => contents.InnerHtml == "Further information")
+				.Should()
+				.NotBeEmpty();
+			Document.QuerySelectorAll("#Further_information_link")
+				.First().Attributes.Where(a => a.Value == "#Further_information")
+				.Should()
+				.NotBeEmpty();
+
+			Document.QuerySelectorAll("#Finances_link")
+				.Where(contents => contents.InnerHtml == "Finances")
+				.Should()				
+				.NotBeEmpty();
+			Document.QuerySelectorAll("#Finances_link")
+				.First().Attributes.Where(a => a.Value == "#Finances")
+				.Should()
+				.NotBeEmpty();
+
+			Document.QuerySelectorAll("#Future_pupil_numbers_link")
+				.Where(contents => contents.InnerHtml == "Future_pupil_numbers")
+				.Should()
+				.NotBeEmpty();
+			Document.QuerySelectorAll("#Future_pupil_numbers_link")
+				.First().Attributes.Where(a => a.Value == "#Future_pupil_numbers")
+				.Should()
+				.NotBeEmpty();
+
+			Document.QuerySelectorAll("#Land_and_buildings_link")
+				.Where(contents => contents.InnerHtml == "Land and buildings")
+				.Should()
+				.NotBeEmpty();
+			Document.QuerySelectorAll("#Land_and_buildings_link")
+				.First().Attributes.Where(a => a.Value == "#Land_and_buildings")
+				.Should()
+				.NotBeEmpty();
+
+			Document.QuerySelectorAll("#Pre-opening_support_grant_link")
+				.Where(contents => contents.InnerHtml == "Pre-opening support grant")
+				.Should()
+				.NotBeEmpty();
+			Document.QuerySelectorAll("#Pre-opening_support_grant_link")
+				.First().Attributes.Where(a => a.Value == "#Pre-opening_support_grant")
+				.Should()
+				.NotBeEmpty();
+
+			Document.QuerySelectorAll("#Consultation_link")
+				.Where(contents => contents.InnerHtml == "Consultation")
+				.Should()
+				.NotBeEmpty();
+			Document.QuerySelectorAll("#Consultation_link")
+				.First().Attributes.Where(a => a.Value == "#Consultation")
+				.Should()
+				.NotBeEmpty();
+
+			Document.QuerySelectorAll("#Declaration_link")
+				.Where(contents => contents.InnerHtml == "Declaration")
+				.Should()
+				.NotBeEmpty();
+			Document.QuerySelectorAll("#Declaration_link")
+				.First().Attributes.Where(a => a.Value == "#Declaration")
+				.Should()
+				.NotBeEmpty();
+		}
```

Comment:
I think we'd have to test that the id is attached to the section header too. I had a stab at it below, but there might be a more direct way to do it
```
			// check the link text and href
			var sectionLink = Document.QuerySelector("#Overview_link");
			sectionLink.TextContent.Should().Be("Overview");
			sectionLink.Attributes
				.Where(a => a.Name == "href")
				.First().Value
				.Should().Be("#Overview");
			// check the link is associated with the correct section header
			Document.QuerySelectorAll("h2")
				.Where(contents => contents.InnerHtml == "Overview")
				.First().Attributes.Where(a => a.Name == "id")
				.First().Value
				.Should().Be("Overview");
```